src/components/rating/rating.tsx

```diff
@@ -227,6 +226,24 @@ export class Rating implements LabelableComponent, FormComponent, InteractiveCom
     this.calciteRatingChange.emit({ value });
   }
 
+  private onKeyboardPressed = (event: KeyboardEvent): void => {
+    if (!this.required && (event.key === "Enter" || event.key === " ")) {
+      //* prevent default event handling only for 'Enter' and 'Space' keys
+      event.preventDefault();
+      this.updateValue(0);
+    }
+  };
+
+  private onFocusChange = (index: number): void => {
+    this.hasFocus = true;
+    // reset input values when the user re-clicks on the input with the focus
```

Comment:
This comment may not be needed if `index` is renamed to something that conveys the same.